clippy_lints/src/deprecated_lints.rs

```diff
@@ -14,6 +14,16 @@ declare_deprecated_lint! {
     "`.extend_from_slice(_)` is a faster way to extend a Vec by a slice"
 }
 
+/// **What it does:** Nothing. This lint has been deprecated.
+///
+/// **Deprecation reason:** `Range::step_by(0)` used to be linted since it's
+/// an infinite iterator, which is better expressed by `iter::repeat`,
+/// but the method has been removed for `Iterator::step_by` which panics
+/// if given a zero
```

Comment:
Nope... `Iterator::step_by` panics if given a zero. I don't think we need to do anything there... well OK, we could move to linting the new method in order to report the error as early as possible... yea let's do that.